apps/system/js/app_usage_metrics.js

```diff
@@ -653,7 +665,15 @@
         oldMetrics.merge(self.metrics);
         self.metrics = oldMetrics;
         self.metrics.save(true);
-      };
+      }
+
+      request.sendRequest({
+        timeout: AUM.REPORT_TIMEOUT,
+        onload: onload,
+        onerror: retry,
```

Comment:
Note that on 4XX, you should not retry the same request - it will fail again :)  It looks like the retry function saves data locally to be rolled in to the next submission, so this may not be an issue.